api/s2n.h

```diff
@@ -1882,6 +1882,18 @@ S2N_API extern uint64_t s2n_connection_get_delay(struct s2n_connection *conn);
  */
 S2N_API extern int s2n_connection_set_cipher_preferences(struct s2n_connection *conn, const char *version);
 
+/**
+ * Signals the connection to do a key_update at the next possible opportunity.
+ * 
+ * @param conn The connection object to trigger the key update on.
+ * @param peer_update_requested Indicates if a key update should also be requested 
+ * of the peer. If false, then only the sending keys of `conn` will be updated. If
+ * true, then the sending keys of conn will be updated AND the peer will be requested
+ * to update their sending key. Note that s2n-tls currently only supports 
+ * `peer_update_requested` being set to `false`.
+ * @returns S2N_SUCCESS on success. S2N_FAILURE on failure
+*/
+S2N_API extern int s2n_connection_request_key_update(struct s2n_connection *conn, bool peer_update_requested);
```

Comment:
Since technically "KeyUpdateRequest" can take any value 0-255 according to the RFC, if we want this to be really future safe we should probably make it an enum or an uint8_t instead of a bool

Reply:
But maybe we're fine since the RFC says that any value other than 0 or 1 results in an error? That's the kind of requirement that's hard to change later :)

Reply:
I don't have a strong opinion on this. I initially had it as an enum but it seemed to be less semantically clear, which is why I'd lean towards the boolean 🤷

Reply:
Synced up offline: I'd feel safer if we use a value that can take on the same values as KeyUpdateRequest from the RFC.